libs/cdk/states/src/resource-registry/resource-registry.state.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Action, State, StateContext } from '@ngxs/store';
+import { tap } from 'rxjs';
+import { AddResource, LoadMarkdown } from './resource-registry.actions';
+import { ResourceEntry, ResourceId, ResourceRegistryModel, ResourceType } from './resource-registry.model';
+
+/**
+ * Resource registry state
+ */
+@State<ResourceRegistryModel>({
+  name: 'resourceRegistry',
+  defaults: {},
+})
+@Injectable()
+export class ResourceRegistryState {
+  /** creates an instance with http */
+  constructor(private http: HttpClient) {}
+
+  /**
+   * Action for adding a resource to state
+   * @param ctx State context
+   * @param param1 addResource action object containing id and entry
+   */
+  @Action(AddResource)
+  addResource(ctx: StateContext<ResourceRegistryModel>, { id, entry }: AddResource): void {
+    this.addResourceEntry(ctx, id, entry);
+  }
+
+  /**
+   * action for loading markdown
+   * @param ctx State context
+   * @param param1 loadMarkdown object containing id and url
+   */
+  @Action(LoadMarkdown)
+  loadMarkdown(ctx: StateContext<ResourceRegistryModel>, { id, url }: LoadMarkdown): void {
+    this.http
+      .get(url, { responseType: 'text' })
+      .pipe(
+        tap((data) => {
+          const entry = { type: ResourceType.Markdown, markdown: data };
+          this.addResourceEntry(ctx, id, entry);
+        })
+      )
+      .subscribe();
```

Comment:
Return the observable instead of subscribing. Ngxs subscribes to returned values automatically